src/components/backend-ai-user-dropdown-menu.ts

```diff
@@ -0,0 +1,523 @@
+      if (!nativeValidity.valid) {
+        if (nativeValidity.valueMissing) {
+          passwordInput.validationMessage = _text('signup.PasswordInputRequired');
+          return {
+            valid: nativeValidity.valid,
+            customError: !nativeValidity.valid
+          };
+        } else {
+          passwordInput.validationMessage = _text('signup.PasswordInvalid');
+          return {
+            valid: nativeValidity.valid,
+            customError: !nativeValidity.valid
+          };
+        }
+      } else {
+        return {
+          valid: nativeValidity.valid,
+          customError: !nativeValidity.valid
+        };
+      }
+    };
+  }
+
+  _validatePassword2() {
+    // TODO define type for custom property
+    const password2Input = this.shadowRoot?.querySelector('#pref-new-password2') as any;
+    password2Input.validityTransform = (newValue, nativeValidity) => {
+      if (!nativeValidity.valid) {
+        if (nativeValidity.valueMissing) {
+          password2Input.validationMessage = _text('signup.PasswordInputRequired');
+          return {
+            valid: nativeValidity.valid,
+            customError: !nativeValidity.valid
+          };
+        } else {
+          password2Input.validationMessage = _text('signup.PasswordInvalid');
+          return {
+            valid: nativeValidity.valid,
+            customError: !nativeValidity.valid
+          };
+        }
+      } else {
+        // custom validation for password input match
+        const passwordInput = this.shadowRoot?.querySelector('#pref-new-password') as TextField;
+        const isMatched = (passwordInput.value === password2Input.value);
+        if (!isMatched) {
+          password2Input.validationMessage = _text('signup.PasswordNotMatched');
+        }
+        return {
+          valid: isMatched,
+          customError: !isMatched
+        };
+      }
+    };
+  }
+
+  /**
+   * Validate User input in password automatically, and show error message if any input error occurs.
+   */
+  _validatePassword() {
+    this._validatePassword1();
+    this._validatePassword2();
+  }
+
+  /**
+   * Update the user information including full_name of user and password
+   */
+  _updateUserInformation() {
+    if (globalThis.backendaiclient.supports('change-user-name')) {
+      this._updateFullname();
+    }
+    this._updateUserPassword();
+    this._refreshUserInfoPanel();
+  }
+  render() {
+    return html`
+      <div class="horizontal flex center layout">
+        <div class="vertical layout center" style="position:relative;right:50px;">
+          <mwc-menu id="dropdown-menu" class="user-menu">
+            ${this.domain !== 'default' && this.domain !== '' ? html`
+            <mwc-list-item class="horizontal layout start center" disabled style="border-bottom:1px solid #ccc;">
+                ${this.domain}
+            </mwc-list-item>
+            ` : html``}
+            <mwc-list-item class="horizontal layout start center">
```

Comment:
This item only shows the user name. But this button is highlighted when hovering. So what do you think about adding `pointer-events:none` to reduce confusion??

Reply:
@Sujin-Kim1 reflected.